schema-registry/client/src/main/java/com/hortonworks/registries/schemaregistry/serde/AbstractSnapshotDeserializer.java

```diff
@@ -144,14 +148,17 @@ public O deserialize(I input,
         byte protocolId = retrieveProtocolId(input);
         SchemaIdVersion schemaIdVersion = retrieveSchemaIdVersion(protocolId, input);
         SchemaVersionInfo schemaVersionInfo = null;
+        SchemaMetadata schemaMetadata = null;
         try {
             schemaVersionInfo = schemaRegistryClient.getSchemaVersionInfo(schemaIdVersion);
+            schemaMetadata = schemaRegistryClient.getSchemaMetadataInfo(schemaVersionInfo.getName()).getSchemaMetadata();
         } catch (SchemaNotFoundException e) {
-            throw new SerDesException(e);
+            throw new RegistryException(e);
```

Comment:
no need to have this as the below catch clause handles this case and throws the same exception.